### PR TITLE
GCS Logging Policy: WIP

### PR DESCRIPTION
**Pull request by SeanTAllen**
Signed-off-by: Sean T. Allen <seanallen@microsoft.com>

### CODE REVIEW

File: pkg/securitypolicy/securitypolicy_marshal.go [hunk 1]
```diff
@@ -91,7 +94,8 @@ func MarshalFragment(
 	containers []*Container,
 	externalProcesses []ExternalProcessConfig,
 	fragments []FragmentConfig) (string, error) {
-	policy, err := newSecurityPolicyInternal(containers, externalProcesses, fragments)
+	// TODO STA: Matt what is the proper thing to do for "marshal fragment" here?
```
Comment (SeanTAllen) [inline pkg/securitypolicy/securitypolicy_marshal.go:97]: @matajoh I'm not sure what to do about marshalling fragments (either here or further up the call chain)

Comment (matajoh) [reply pkg/securitypolicy/securitypolicy_marshal.go:97]: Agreed. I think we probably need to separate out the paths entirely and sensibly limit what gets emitted from a policy for fragments. We may need a different type (i.e. a `securityPolicyFragment` or similar which has a subset of the members of `securityPolicyInternal`)

**Comment by SeanTAllen** [comment]
@anmaxvl here's my idea that i went over with you yesterday. let's discuss.

**Comment by SeanTAllen** [comment]
@matajoh assuming that Maksim is ok with this approach, can you give a review to make sure I didn't mess anything up? I think that naming could be improved but I wanted to focus on the approach and worry about names if the approach was agreed upon.

File: cmd/gcs/main.go [hunk 2]
```diff
@@ -194,6 +195,9 @@ func main() {
 		false,
 		"If true do not run chronyd time synchronization service inside the UVM")
 	scrubLogs := flag.Bool("scrub-logs", false, "If true, scrub potentially sensitive information from logging")
+	initialPolicyStance := flag.String("initial-policy-stance",
+		"deny",
```
Comment (KenGordon) [inline cmd/gcs/main.go:199]: Either the script will have to set this to deny explicitly or create_lcow will have to explicitly set it to allow. I think it is better to have the script do the explicit thing for the confidential case that to rely on the default remaining "deny" and not rotting to allow when someone mistakenly changes it to default the other way.

Comment (SeanTAllen) [reply cmd/gcs/main.go:199]: yeah, "deny" as the default maintains the status quo of using closed door if no flag is given. It changes the status quo in that some logging will now be denied.

Comment (KenGordon) [reply cmd/gcs/main.go:199]: I guess we just set it explicitly in both startup.sh and in create_lcow.go where is creates the kernel command line (via execCmdArgs, initArgs and eventually kernelArgs)

Comment (anmaxvl) [reply cmd/gcs/main.go:199]: not sure if we care about deny in non-confidential case.

File: internal/guest/runtime/hcsv2/uvm.go [hunk 3]
```diff
@@ -63,16 +65,20 @@ type Host struct {
 	securityPolicyEnforcer    securitypolicy.SecurityPolicyEnforcer
 	securityPolicyEnforcerSet bool
 	uvmReferenceInfo          string
+
+	// logging target
+	loggingTargetFile *os.File
```
Comment (anmaxvl) [inline internal/guest/runtime/hcsv2/uvm.go:70]: `logrus.SetOuptut` takes an `io.Writer`, so maybe this should be a `logWriter io.Writer`. This way the caller can pass something that implements the interface.

File: internal/guest/runtime/hcsv2/uvm.go [hunk 4]
```diff
@@ -102,6 +108,19 @@ func (h *Host) SetConfidentialUVMOptions(ctx context.Context, r *guestresource.L
...
 	}
 
+	// This is one of two points at which we might change our logging.
+	// At this time, we now have a policy and can determine what the policy
+	// author put as policy around runtime logging.
+	// The other point is on startup where we take a flag to set the default
+	// policy enforcer to use before a policy arrives. After that flag is set,
+	// we use the enforcer in question to set up logging as well.
+	err = p.EnforceRuntimeLoggingPolicy()
+	if err == nil {
```
Comment (anmaxvl) [inline internal/guest/runtime/hcsv2/uvm.go:118]: nit: `if err := p.EnforceRuntimeLoggingPolicy(); err != nil {} else {}`

**Pull request closed without merging**